chainercv/datasets/sbd/sbd_instance_segmentation_dataset.py

```diff
@@ -0,0 +1,123 @@
+
+    """
+
+    def __init__(self, data_dir='auto', split='train'):
+        _check_available()
+
+        if split not in ['train', 'trainval', 'val']:
+            raise ValueError(
+                'please pick split from \'train\', \'trainval\', \'val\'')
+
+        if data_dir == 'auto':
+            data_dir = sbd_utils.get_sbd()
+
+        id_list_file = os.path.join(
+            data_dir, '{}_voc2012.txt'.format(split))
+        self.ids = [id_.strip() for id_ in open(id_list_file)]
+
+        self.data_dir = data_dir
+
+    def __len__(self):
+        return len(self.ids)
+
+    def get_example(self, i):
+        """Returns the i-th example.
+
+        Returns a color image, bounding boxes, masks and labels. The color
+        image is in CHW format.
+
+        Args:
+            i (int): The index of the example.
+
+        Returns:
+            tuple of color image, bounding boxes, masks and labels whose
+            shapes are (3, H, W), (R, 4), (R, H, W), (R, ) respectively.
+            H and W are height and width of the images, and R is the number
+            of objects in the image. The dtype of the color image and
+            the bounding boxes are :obj:`numpy.float32`, that of the
+            masks are :obj: `bool`, and that of the labels are
+            :obj:`numpy.int32`.
+
+        """
+        data_id = self.ids[i]
+        img_file = os.path.join(
+            self.data_dir, 'img', data_id + '.jpg')
+        img = read_image(img_file, color=True)
+        seg_img, ins_img = self._load_seg_ins(data_id)
+        bbox, mask, label = self._prepare_data(seg_img, ins_img)
+        return img, bbox, mask, label
+
+    def _load_seg_ins(self, data_id):
+        seg_file = os.path.join(
+            self.data_dir, 'cls', data_id + '.mat')
+        ins_file = os.path.join(
+            self.data_dir, 'inst', data_id + '.mat')
+        seg_mat = scipy.io.loadmat(seg_file)
+        seg_img = seg_mat['GTcls']['Segmentation'][0][0].astype(np.int32)
+        ins_mat = scipy.io.loadmat(ins_file)
+        ins_img = ins_mat['GTinst']['Segmentation'][0][0].astype(np.int32)
+        ins_img[ins_img == 0] = -1
+        ins_img[ins_img == 255] = -1
+        return seg_img, ins_img
+
+    def _prepare_data(self, seg_img, ins_img):
+        bbox = []
+        mask = []
+        label = []
+        instances = np.unique(ins_img)
+        for inst_id in instances[instances != -1]:
+            mask_inst = ins_img == inst_id
+            instance_class = np.unique(seg_img[mask_inst])[0] - 1
+
+            assert inst_id != -1
+            assert instance_class != -1
+
+            where = np.argwhere(mask_inst)
+            (y1, x1), (y2, x2) = where.min(0), where.max(0) + 1
+
+            bbox.append((y1, x1, y2, x2))
+            mask.append(mask_inst)
+            label.append(instance_class)
+        bbox = np.array(bbox).astype(np.float32)
+        mask = np.array(mask).astype(bool)
```

Comment:
Please use `np.bool` instead of `bool`.
I prefer not to use Python's builit-in types for casting numpy arrays because it is inconsistent.
We do not do `array.astype(float)`, but do `array.astype(np.float64)`.

Reply:
I think `np.bool` is same as `bool`.
In numpy doc, there is no `np.bool` dtype officially.
https://docs.scipy.org/doc/numpy-1.13.0/user/basics.types.html

```python
In [1]: import numpy as np

In [2]: np.bool
Out[2]: bool

In [3]: bool
Out[3]: bool

In [4]: bool == np.bool
Out[4]: True
```